Kernel/Devices/USB/USBDevice.h

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright (c) 2021, Jesse Buhagiar <jooster669@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <AK/Types.h>
+#include <Kernel/Devices/USB/USBPipe.h>
+
+namespace Kernel::USB {
+
+//
+// Some nice info from FTDI on device enumeration and how some of this
+// glues together:
+//
+// https://www.ftdichip.com/Support/Documents/TechnicalNotes/TN_113_Simplified%20Description%20of%20USB%20Device%20Enumeration.pdf
+class USBDevice {
+public:
+    enum class PortNumber : u8 {
+        Port1 = 0,
+        Port2
+    };
+
+    enum class DeviceSpeed : u8 {
+        FullSpeed = 0,
+        LowSpeed
+    };
+
+public:
+    static void create(PortNumber port, DeviceSpeed speed);
+    static USBDevice* get(PortNumber port);
```

Comment:
Please omit parameter names when they are obvious from context and/or the type names.